sklearn/linear_model/tests/test_sgd.py

```diff
@@ -2144,3 +2165,21 @@ def test_loss_squared_loss_deprecated(Estimator):
         assert_allclose(est1.predict_proba(X), est2.predict_proba(X))
     else:
         assert_allclose(est1.predict(X), est2.predict(X))
+
+
+def test_quantile():
+    X, y = datasets.make_regression(
+        n_samples=1000, n_features=10, n_informative=9, random_state=1234
+    )
+    from .._quantile import QuantileRegressor
+
+    benchmark = QuantileRegressor(alpha=1e-6, solver="highs").fit(X, y).score(X, y)
+
+    est = SGDRegressor(
+        alpha=1e-5,
+        loss="pinball",
+        quantile=0.5,
+        max_iter=50,
+        random_state=42,
+    ).fit(X, y)
+    assert est.score(X, y) + 0.2 > benchmark
```

Comment:
This really a flaky test here. We should have really close score.

Reply:
basically you want to check that the score of sgd - score of the qr is close to zero.
I would as well check the parameter of the coefficient and intercept.

We might need to set the `learning_rate` and the tolerance because it might be something quite sensitive with SGD

Reply:
Thanks. cooked up an example with similar coefs and performance.